GVFS/GVFS.FunctionalTests/Tests/MultiEnlistmentTests/SharedCacheTests.cs

```diff
@@ -60,6 +60,28 @@ public void SecondCloneDoesNotDownloadAdditionalObjects()
                 .ShouldMatchInOrder(allObjects);
         }
 
+        [TestCase]
+        public void SecondCloneSucceedsWithMissingTrees()
+        {
+            GVFSFunctionalTestEnlistment enlistment1 = this.CloneAndMountEnlistment();
+            File.ReadAllText(Path.Combine(enlistment1.RepoRoot, WellKnownFile));
+
+            this.AlternatesFileShouldHaveGitObjectsRoot(enlistment1);
+
+            string packDir = Path.Combine(enlistment1.GetObjectRoot(this.fileSystem), "pack");
+            string[] packDirFiles = Directory.EnumerateFiles(packDir, "*", SearchOption.AllDirectories).ToArray();
+            for (int i = 0; i < packDirFiles.Length; i++)
+            {
+                this.fileSystem.DeleteFile(Path.Combine(packDir, packDirFiles[i]));

```

Comment:
Technically, this is changing the behavior of all the tests that follow. Can we do a prefetch at the end of this test to get back to the original state?